sdk/core/Azure.Core/src/Operation.cs

```diff
@@ -90,6 +90,25 @@ public virtual async ValueTask<Response> WaitForCompletionResponseAsync(TimeSpan
             return await poller.WaitForCompletionResponseAsync(this, pollingInterval, cancellationToken).ConfigureAwait(false);
         }
 
+        /// <summary>
+        /// Periodically calls the server till the long-running operation completes.
+        /// </summary>
+        /// <param name="delay">
+        /// The interval between status requests to the server.
+        /// The interval can change based on information returned from the server.
+        /// For example, the server might communicate to the client that there is not reason to poll for status change sooner than some time.
+        /// </param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> used for the periodical service calls.</param>
+        /// <returns>The last HTTP response received from the server.</returns>
+        /// <remarks>
+        /// This method will periodically call UpdateStatusAsync till HasCompleted is true, then return the final response of the operation.
+        /// </remarks>
+        public virtual async ValueTask<Response> WaitForCompletionResponseAsync(Delay delay, CancellationToken cancellationToken = default)
```

Comment:
I think we are still missing `WaitForCompletionResponse`?

Reply:
Woops - added.